### PR TITLE
HBASE-24151 The master server aborted for IllegalThreadStateException

### PR DESCRIPTION
**Pull request by Gkkkk302**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   4m 33s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-2.2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 47s |  branch-2.2 passed  |
| +1 :green_heart: |  compile  |   0m 25s |  branch-2.2 passed  |
| +1 :green_heart: |  checkstyle  |   0m 13s |  branch-2.2 passed  |
| +1 :green_heart: |  shadedjars  |   4m 27s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 22s |  branch-2.2 passed  |
| +0 :ok: |  spotbugs  |   0m 48s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   0m 46s |  branch-2.2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 15s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 25s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 25s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 14s |  hbase-rsgroup: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 26s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  23m 29s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 2.10.0 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 21s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   0m 55s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   9m 16s |  hbase-rsgroup in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 12s |  The patch does not generate ASF License warnings.  |
|  |   |  67m 59s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1465/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1465 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 4281e36cf257 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1465/out/precommit/personality/provided.sh |
| git revision | branch-2.2 / 1d912790d5 |
| Default Java | 1.8.0_181 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1465/1/artifact/out/diff-checkstyle-hbase-rsgroup.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1465/1/testReport/ |
| Max. process+thread count | 4006 (vs. ulimit of 10000) |
| modules | C: hbase-rsgroup U: hbase-rsgroup |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1465/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-rsgroup/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupBasedLoadBalancer.java [hunk 1]
```diff
@@ -356,6 +357,10 @@ public void initialize() throws HBaseIOException {
     // Create the balancer
     Class<? extends LoadBalancer> balancerKlass = config.getClass(HBASE_RSGROUP_LOADBALANCER_CLASS,
         StochasticLoadBalancer.class, LoadBalancer.class);
+    if (balancerKlass.equals(this.getClass())) {
+      LOG.warn("The internal balancer of RSGroupBasedLoadBalancer cannot be itself, falling back to the default LoadBalancer class");
```
Comment (virajjasani) [inline hbase-rsgroup/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupBasedLoadBalancer.java:361]: line len > 100 chars

**Comment by saintstack** [comment]
Is this patch for branch-2.1 and branch-2.2 only @Gkkkk302 ? What about branch-2.3 and master? Thank you.

**Comment by Gkkkk302** [comment]
> Is this patch for branch-2.1 and branch-2.2 only @Gkkkk302 ? What about branch-2.3 and master? Thank you.
This patch should also apply to branch-2.3, I will submit a pr for branch-2.3, the master branch seems to have been repaired

**Comment by Gkkkk302** [comment]
@saintstack  Thank you for your comments.This patch should also apply to branch-2.3, I will submit a pr for branch-2.3, the master branch seems to have been repaired.

File: hbase-rsgroup/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupBasedLoadBalancer.java (hunk 1 above)
Comment (Gkkkk302) [reply hbase-rsgroup/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupBasedLoadBalancer.java]: @virajjasani Thanks for review, I resubmitted a PR for branch-2.2

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m 25s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-2.2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   7m 12s |  branch-2.2 passed  |
| +1 :green_heart: |  compile  |   0m 30s |  branch-2.2 passed  |
| +1 :green_heart: |  checkstyle  |   0m 15s |  branch-2.2 passed  |
| +1 :green_heart: |  shadedjars  |   5m 23s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 27s |  branch-2.2 passed  |
| +0 :ok: |  spotbugs  |   1m  5s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m  2s |  branch-2.2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 59s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 34s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 34s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 15s |  hbase-rsgroup: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   5m 25s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  27m 54s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 2.10.0 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 24s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   1m  5s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 58s |  hbase-rsgroup in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 15s |  The patch does not generate ASF License warnings.  |
|  |   |  72m 28s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1465/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1465 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 1a0a2f28228c 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1465/out/precommit/personality/provided.sh |
| git revision | branch-2.2 / 653fb90270 |
| Default Java | 1.8.0_181 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1465/2/artifact/out/diff-checkstyle-hbase-rsgroup.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1465/2/testReport/ |
| Max. process+thread count | 4245 (vs. ulimit of 10000) |
| modules | C: hbase-rsgroup U: hbase-rsgroup |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1465/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 53s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-2.2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   7m  6s |  branch-2.2 passed  |
| +1 :green_heart: |  compile  |   0m 31s |  branch-2.2 passed  |
| +1 :green_heart: |  checkstyle  |   0m 16s |  branch-2.2 passed  |
| +1 :green_heart: |  shadedjars  |   5m 30s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 26s |  branch-2.2 passed  |
| +0 :ok: |  spotbugs  |   1m  3s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m  0s |  branch-2.2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 19s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 31s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 31s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 15s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   5m 52s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  25m 35s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 2.10.0 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 22s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   0m 52s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 38s |  hbase-rsgroup in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 14s |  The patch does not generate ASF License warnings.  |
|  |   |  65m 47s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1465/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1465 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 5b3a2bb9af89 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1465/out/precommit/personality/provided.sh |
| git revision | branch-2.2 / 653fb90270 |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1465/3/testReport/ |
| Max. process+thread count | 4207 (vs. ulimit of 10000) |
| modules | C: hbase-rsgroup U: hbase-rsgroup |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1465/3/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by HorizonNet** [comment]
Will this be covered in #1508?

**Comment by Gkkkk302** [comment]
yes

**Comment by HorizonNet** [comment]
Ok. Do we want to close this one?

**Comment by virajjasani** [comment]
> Ok. Do we want to close this one?

Sure

**Pull request closed without merging**